kafka-connect-aws-s3/src/test/scala/io/lenses/streamreactor/connect/aws/s3/auth/AwsContextCreatorTest.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Lenses.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.lenses.streamreactor.connect.aws.s3.auth
+
+import com.amazonaws.auth.{AWSCredentials, AWSCredentialsProvider}
+import io.lenses.streamreactor.connect.aws.s3.config.{AuthMode, S3Config}
+import org.jclouds.blobstore.BlobStoreContext
+import org.jclouds.providers.ProviderMetadata
+import org.jclouds.rest.internal.ApiContextImpl
+import org.mockito.MockitoSugar
+import org.scalatest.BeforeAndAfter
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class AwsContextCreatorTest extends AnyFlatSpec with MockitoSugar with Matchers with BeforeAndAfter {
+
+  private val AWS_ACCESS_KEY_ID = "awsAccessKeyId"
+
+  private val awsCredentials = mock[AWSCredentials]
+  private val credentialsProvider = mock[AWSCredentialsProvider]
+  private val credentialsProviderFn = mock[() => AWSCredentialsProvider]
+
+  private val target = new AwsContextCreator(credentialsProviderFn)
+
+  before {
+    reset(awsCredentials, credentialsProvider, credentialsProviderFn)
+  }
+
+  "fromConfig" should "use default credentials provider when auth mode is 'Default'" in {
```

Comment:
this is just syntax. you can do

`
"fromConfig" should {
 "use default credentials provider when auth mode is 'Default'" in {...}
  "use supplied credentials when auth mode is 'Credentials'" in {...}
}
`

Reply:
Perhaps something to address at a later date, as all of the tests in the S3 sink project are using AnyFlatSpec and wouldn't want to introduce inconsistencies